### PR TITLE
docs: vector type

### PR DESCRIPTION
**Pull request by zhongzc**
## What's Changed in this PR

Vector type related docs.

## Checklist

- [ ] Please confirm that all corresponding versions of the documents have been revised.
- [ ] Please ensure that the content in `sidebars.ts` matches the current document structure when you changed the document structure.
- [ ] This change requires follow-up update in localized docs.

### CODE REVIEW

File: docs/reference/sql/functions/vector.md [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+Example:
+
+```sql
+SELECT l2sq_distance('[1.0, 2.0, 3.0]', '[2.0, 1.0, 4.0]');
+```
+
+Details:
+
+* Input: Two vectors with consistent dimensions (as literals or column values).  
+* Output: A scalar value of type `f64`.
```
Comment (nicecui) [inline docs/reference/sql/functions/vector.md:30]:
```suggestion
* Output: A scalar value of type `Float64`.  
```

File: docs/reference/sql/functions/vector.md [hunk 2]
```diff
@@ -0,0 +1,64 @@
...
+
+Example:
+
+```sql
+SELECT l2sq_distance('[1.0, 2.0, 3.0]', '[2.0, 1.0, 4.0]');
+```
+
+Details:
+
+* Input: Two vectors with consistent dimensions (as literals or column values).
```
Comment (nicecui) [inline docs/reference/sql/functions/vector.md:29]: as mentioned above, does the column values mean column names?

File: docs/reference/sql/functions/vector.md [hunk 3]
```diff
@@ -0,0 +1,64 @@
...
+Example:
+
+```sql
+SELECT cos_distance('[1.0, 2.0, 3.0]', '[2.0, 1.0, 4.0]');
+```
+
+Details:
+
+* Input: Two vectors with consistent dimensions (as literals or column values).  
+* Output: A scalar value of type `f64`.
```
Comment (nicecui) [inline docs/reference/sql/functions/vector.md:47]:
```suggestion
* Output: A scalar value of type `Float64`.  
```

File: docs/reference/sql/functions/vector.md [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+Example:
+
+```sql
+SELECT dot_product('[1.0, 2.0, 3.0]', '[2.0, 1.0, 4.0]');
+```
+
+Details:
+
+* Input: Two vectors with consistent dimensions (as literals or column values).  
+* Output: A scalar value of type `f64`.
```
Comment (nicecui) [inline docs/reference/sql/functions/vector.md:64]:
```suggestion
* Output: A scalar value of type `Float64`.  
```

**Review by nicecui — COMMENTED** [review]
The document structure is LGTM.
Some details need to be changed, please also modify the zh docs according to the comments, thanks

**Review by nicecui — COMMENTED** [review]
LGTM, I recommend using `Vector Store` in the navigation to avoid confusing with the `vector` tool.
please add the zh title to `/i18n/zh/docusaurus-plugin-content-docs/current.json`

**Review by nicecui — APPROVED** [review]

**Review by fengjiachun — APPROVED** [review]
LGTM

**Pull request merged**